ICS 2: Consensus Verification (#20)
cosmos/ics: spec/ics-2-consensus-verification/VERIFIER.md

# Verifier

> Full nodes are procedures who run the all `[Message]` according to the

Reviewer: ```suggestion
Full nodes are procedures which process a list of messages, `[Message]`, according to the
```

> `Consensus`. The lightclient `Verifier` does not runs all `[Message]`, but

Reviewer: ```suggestion
`Consensus` algorithm. The lightclient `Verifier` does not process all messages, but
```

> it uses the `Consensus`'s security assumption to fast-forward the verification

Reviewer: ```suggestion
it uses artifacts produced by execution of the `Consensus` algorithm, along with well-forumated security assumptions, to verify parts of a consensus execution transcript
```

> process. The `Verifier` MUST work identical with a full node while the

Reviewer: ```suggestion
process. The `Verifier` MUST work identically to a full node, given that the
```

> security assumptions of `Consensus` are preserved. Which means, if and only

Reviewer: ```suggestion
security assumptions of `Consensus` are preserved. This means that if and only
```

if a full node accepts the new `Header` given a `ConsensusState` and

> `[Message]`, then the `Verifier` also accepts it.

Reviewer: ```suggestion
`[Message]`, will the `Verifier` then also accept it.
```

(alternatively, reorder the sentence)

> ## Definitions

Reviewer 2: Wait I'm confused there is a definition section above. Why are there two definitions sections in the docs?

Author: Oh they are different files, the verifier one is specifically about lightclient verifier internal, where readme is the interface part for the verifier.
